Type hints + simple Pytest + higher precision serial

LelsersLasers opened the pull request:
Bonus: an (untested) simple test for collector plate for the old boards and breadboard/simple wires

LelsersLasers committed:
Fixing things being double tabbed

LelsersLasers committed:
Fixing string formatting to avoid invalid escape characters

LelsersLasers committed:
Fixing import error

LelsersLasers committed:
unix file slash directions

LelsersLasers committed:
Creating a super simple pytest example

LelsersLasers committed:
Trying to convert to pytest

Not sure if it is working because I don't understand how to actually test the code without all the boards

LelsersLasers committed:
Testing check.almost_equal()

LelsersLasers committed:
Bugfix: setting up all of hil in the pytest fixture

LelsersLasers committed:
Section dividers

LelsersLasers committed:
Trying abox pytest

LelsersLasers committed:
Don't need to call the functions when using pytest

LelsersLasers committed:
Can pass no arguments to run on all files

LelsersLasers committed:
Readability/format changes

LelsersLasers committed:
Python decorator for logging HIL status

LelsersLasers committed:
Typo in abox text config load

Copied the charger template, forgot to switch it out

LelsersLasers committed:
Trying Pytest collector

LelsersLasers committed:
Folder rename

LelsersLasers committed:
< now correctly <= for the left bound

LelsersLasers committed:
Trying Pytest main base

LelsersLasers committed:
Confirming that non "hil" params are not effected

LelsersLasers committed:
Trying Pyest for main sdc

LelsersLasers committed:
Typo: these tests don't read power

Result of copying and pasting without looking :shrug:

LelsersLasers committed:
Trying Pytest system

LelsersLasers committed:
Trying Pytest test

Kinda weird because it makes no calls to HIL.check()

LelsersLasers committed:
Separating new Pytest scripts and old ones

Can use this commit to see all the way normal tests were converted to pytest ones by looking at the modified files in scripts/

LelsersLasers committed:
Commenting out old code for in hil.py testing

LelsersLasers committed:
Type hinting return types and instance variables

LelsersLasers committed:
Type hinting hil.py

LelsersLasers committed:
Type hinting pin_mapper.py

LelsersLasers committed:
Type hinting utils

LelsersLasers committed:
Trying to improve imports?

Makes the errors go away in VSCode, but there was probably a reason it was the way it was :shrug:

LelsersLasers committed:
Type hinting component.py

LelsersLasers committed:
Type hinting hil_device.py

LelsersLasers committed:
Type hinting serial_manager.py

LelsersLasers committed:
Adding list of file which need to be type hinted

LelsersLasers committed:
Commenting out unused imports

LelsersLasers committed:
Almost done type hinting client.py

missing types on queue items

LelsersLasers committed:
Finishing type hints for client.py

Added type hints send/msg types

LelsersLasers committed:
Update README.md

LelsersLasers committed:
Almost done type hinting can_bus.py

LelsersLasers committed:
Finishing type hinting can_bus.py (I think)

LelsersLasers committed:
Updating list of completed files

LelsersLasers committed:
Type hinting daq_protocal.py -> DAQPin()

LelsersLasers committed:
(Mostly) type hinting daq_protocol.py -> DaqProtocol

LelsersLasers committed:
(Mostly) type hinting daq_protocol.py -> DAQVariable

LelsersLasers committed:
Finishing type hinting daq_protocol.py

LelsersLasers committed:
Done with all files

LelsersLasers committed:
No longer need the TODO

LelsersLasers committed:
Trying to understand the structure of utils.signals

Thanks ChatGPT!

LelsersLasers committed:
Fixing utils.signals outline/structure

ChatGPT was wrong

LelsersLasers committed:
Type hinting/commenting out globals

LelsersLasers committed:
Fixing type hint (BusSignal from util.signals not str)

LelsersLasers committed:
Fixing typo

LelsersLasers committed:
Removing duplicate line

LelsersLasers committed:
Folder name must match ino name (case sensitive)

testbench -> TestBench.ino

LelsersLasers committed:
Trying to add DFRobot_MCP4725 driver

Source: https://github.com/DFRobot/DFRobot_MCP4725

LelsersLasers committed:
DFRobot_MCP4725: local #include instead of library global #include

LelsersLasers committed:
Indentation: 2 spaces -> tabs

With tabs, each person can choose the display size

LelsersLasers committed:
Adding README to the repo for how I think it could be run

LelsersLasers committed:
typo fix (missing end quote)

LelsersLasers committed:
Fixing misnamed test script

LelsersLasers committed:
type hints fix: global + type hint = bad

Now:
- Declared in global scope with type hint
- Simply assigned in utils.initGlobals()

LelsersLasers committed:
Fixing imports. Everything relative to ./ instead of ./hil now

- Only fixed for test_abox
- Broke 2 type hints in the process to solve the circular  dependency

LelsersLasers committed:
Fixing path for all pytest scripts

LelsersLasers committed:
using typing.TYPE_CHECKING to avoid circular dependencies

LelsersLasers committed:
Removing unneeded logging

LelsersLasers committed:
Firmware path for Millan Kumar

LelsersLasers committed:
Correctly initializing can for test_test.py

LelsersLasers committed:
Debug print for while CanBus init fails

LelsersLasers committed:
Pytest for config_dash.py

LelsersLasers committed:
"config_dash" -> "test_dash" for naming consistency

LelsersLasers committed:
Correctly lower casing scripts folder

LelsersLasers committed:
Adding note about firmware path

LelsersLasers committed:
Format fix

LelsersLasers committed:
A note about inputs vs output naming

Likely obvious if you are competent, but I am not :shrug:

LelsersLasers committed:
Slightly cleaning code and adding prints

LelsersLasers committed:
Adding a comment explaining the bit and-ing

LelsersLasers committed:
Only one arduino so id = 1??

LelsersLasers committed:
Deleting old script code

LelsersLasers committed:
Cleaning prints

LelsersLasers committed:
Can't use "input" with Pytest

LelsersLasers committed:
Improving teardown print statements

LelsersLasers committed:
Improving prints

LelsersLasers committed:
test_collector but no pytest so can wait using input()

Also has some improved print formatting

LelsersLasers committed:
Type hinting new test

LelsersLasers committed:
Improving test output + tabs

LelsersLasers committed:
Test to toggle a DO to make sure things work

LelsersLasers committed:
idk why the id is 2 now

LelsersLasers committed:
Simple Arduino program that toggles DO for pin 4

LelsersLasers committed:
Debug print

LelsersLasers committed:
Making the arduino LED reflect the pin high/low

LelsersLasers committed:
Python HIL can set Arduino pins high/low

LelsersLasers committed:
HIL Micro led matches last GPIO write

LelsersLasers committed:
Currently using an Arduino micro

LelsersLasers committed:
We don't need DAC features rn (I think)

LelsersLasers committed:
renaming to millan.py

LelsersLasers committed:
Working!: Fully custom DO and DI HIL test (without the board)

LelsersLasers committed:
Pytest version of custom test

LelsersLasers committed:
Trying to test "AI" mode (unsuccessful?)

LelsersLasers committed:
Testing DI and AI at the same time

LelsersLasers committed:
Better comment for Arduino discovery

LelsersLasers committed:
Tester id = 1 (from 2)

LelsersLasers committed:
Creating files to setup a breadboard Collector test

LelsersLasers committed:
Trying to setup a breadboard test for Collector Plate

LelsersLasers committed:
Indentation changes

LelsersLasers committed:
Commenting out unneeded variable declaration

LelsersLasers committed:
Fixing board name for simple collector test

LelsersLasers committed:
Changing print statements slightly

LelsersLasers committed:
New config for breadboard + 12 bit dac

LelsersLasers committed:
Renaming folder

LelsersLasers committed:
Deleting old test I made while learning

LelsersLasers committed:
Writing the python test (and it should dictate how the net map is made)

LelsersLasers committed:
mil_dac: maybe net map and config (untested)

LelsersLasers committed:
Slightly improving breadboard 12-bit DAC test

LelsersLasers committed:
I think I figured out how to use the DAC and maybe a valid pin mapping for the board

LelsersLasers committed:
Renaming for Arduino IDE

LelsersLasers committed:
Fixing file name

LelsersLasers committed:
DX pins don't work, AX pins work fine for DI/DO at least

LelsersLasers committed:
Successfully running DI/DO on AI pins

LelsersLasers committed:
Fixing typo

LelsersLasers committed:
Can write 0 voltage :shrug:

LelsersLasers committed:
DAC works! The math for voltage to DAC is wrong

Also: you can only write: 0-255 because it is a char over serial, but then it defeats the purpose of a 12-bit dac

LelsersLasers committed:
DAC is effectively 8-bit when sent over serial

LelsersLasers committed:
Flipping addresses for DACs

More intuitive. The first DAC is the one with the "normal"/not modified address

LelsersLasers committed:
Writing a test for the RLY

LelsersLasers committed:
Adding last year's "pcb" device config for DI pins for comparison

LelsersLasers committed:
Improving RLY test to take DO input

LelsersLasers committed:
Renaming test

LelsersLasers committed:
WRITE_PWM command/device type

LelsersLasers committed:
Adding PWM pins to the device config

LelsersLasers committed:
Reorganizing files

LelsersLasers committed:
net_map and config for all the possible pins; need to update test

LelsersLasers committed:
Updating test to match net_map and config changes

LelsersLasers committed:
Test for pwm. Runs all 4 PWMs

LelsersLasers committed:
Testing both DACs

LelsersLasers committed:
Improving print formating

LelsersLasers committed:
Formatting changes

LelsersLasers committed:
Untested: HIL python can now send data of arbitrary data sizes

Instead of being locked to [command, pin, value], it is now [command, ...] where the amount of things is specified by an array that is parallel to GpioCommand.
100% untested

LelsersLasers committed:
Theoretically: DACS can now function as 12-bit

LelsersLasers committed:
Fixing typo (code compiles)

LelsersLasers committed:
Built in LED switches state whenever a command finishes

LelsersLasers committed:
Merge pull request #9 from PurdueElectricRacing/feature/millan/arbitraryDataSize

feature/millan/arbitraryData Size

LelsersLasers committed:
There are only 2 DACS  to test (typo from 4)

LelsersLasers committed:
Rewriting collector plate to use RLY1 as a DO

Need 4 DO and 1 AI, but only AX headers can do either, so one DO is mimicked from a RLY

LelsersLasers committed:
Finally testing DX pins (3-7 work, 8 not usable, 1 & 2 don't work)

LelsersLasers committed:
Can't use LED_BUILTIN because it is tied to 13

LelsersLasers committed:
RLY3 and RLY4 are flipped

LelsersLasers committed:
AX pins shouldn't be used for DO because they are not pulled down

LelsersLasers committed:
The relays now work

LelsersLasers committed:
Reorganzing

LelsersLasers committed:
Removing old test files

LelsersLasers committed:
Delete config_millan.json

LelsersLasers committed:
Delete millan_net_map.csv

LelsersLasers committed:
Deleting old files

LelsersLasers committed:
Adjusting the simple col test for the inverted relay

AdityaAsGithub commented on TestBench/DFRobot_MCP4725.cpp:
Try to get regular include working

LelsersLasers replied:
Thought: It was initially set up as an Arduino library, but does it make sense to leave it this way so A) it is similar to the way the digipot library is included and B) so minimal setup is needed to flash the code? However, in theory, people shouldn't be needing to reflash the code once everything works (and I might just be struggling to use the Arduino IDE),

AdityaAsGithub commented on _old/ArduinoFlashTest/ArduinoFlashTest.ino:
Remove this

AdityaAsGithub commented on _old/SimplePyTestExample/fake_test.py:
Remove this

AdityaAsGithub commented on _old/SimplePyTestExample/test.sh:
Remove this

AdityaAsGithub commented on configurations/config_collector_bench.json:
Undo this change

AdityaAsGithub commented on hil/communication/can_bus.py:
```diff
@@ -1,61 +1,66 @@
 import threading
 import numpy as np
-import math
+# import math
```
Delete the line

AdityaAsGithub commented on hil/communication/can_bus.py:
```diff
@@ -353,68 +359,88 @@ class BusSignal():
 
-    def update(self, val, timestamp):
+    def update(self, val: int, timestamp: float) -> None:
+        # TODO: not sure if the type hints are correct
```
ehhhh im sure fiiiiine

AdityaAsGithub commented on hil/communication/can_bus.py:
```diff
@@ -353,68 +359,88 @@ class BusSignal():
     @property
-    def curr_val(self):
+    def curr_val(self) -> int:
+        # TODO: not sure if the type hints are correct
```
If you don't investigate they arent wrong

AdityaAsGithub commented on hil/communication/daq_protocol.py:
```diff
@@ -46,27 +47,56 @@
+        offset: int = 0
+    ):
+        super(DAQVariable, self).__init__(
+            bus_name,
+            node_name,
+            msg_name,
+            sig_name,
+            dtype,
```
If you feel so inclined

AdityaAsGithub commented on hil/communication/daq_protocol.py:
```diff
@@ -89,7 +119,9 @@ def fromDAQVar(id, var, node, bus):
-    def fromDAQFileVar(id, var, file_name, file_lbl, node, bus):
+    @classmethod
+    def fromDAQFileVar(cls, id: int, var: dict, file_name: str, file_lbl: str, node: dict, bus: dict) -> DAQVariable:
+        # TODO: not sure if the type hints are correct
```
bruh

AdityaAsGithub commented on hil/communication/daq_protocol.py:
```diff
@@ -108,15 +140,17 @@ def fromDAQFileVar(id, var, file_name, file_lbl, node, bus):
-    def valueSendable(self, val):
+    def valueSendable(self, val: float) -> bool:
+        # TODO: not sure if the type hint is correct
         # TODO: check max and min from json config
```
MOre Type hints TODO

AdityaAsGithub commented on hil/communication/daq_protocol.py:
```diff
@@ -129,27 +163,29 @@ def valueSendable(self, val):
 
-    def getSendValue(self, val):
+    def getSendValue(self, val: float) -> float | bool:
+        # TODO: not sure if the type hint is correct
```
thype hint

AdityaAsGithub commented on hil/communication/daq_protocol.py:
```diff
@@ -172,32 +209,35 @@ def state(self, s):
+        self.can_bus: CanBus = bus
         self.can_bus.handle_daq_msg = self.handleDaqMsg
 
+        # self.can_bus.new_msg_sig.connect(self.handleDaqMsg)
```
think you can delete

AdityaAsGithub commented on hil/communication/daq_protocol.py:
We should try connecting to a cannable testbench just to make sure we still can

AdityaAsGithub commented on hil/components/component.py:
```diff
@@ -1,4 +1,9 @@
+
+import hil.utils as utils
+if TYPE_CHECKING:
+    from hil.hil import HIL
```
idk man I'm thinking about this @AdityaAsGithub

AdityaAsGithub replied:
idk man I didnt' think about this yet @AdityaAsGithub

AdityaAsGithub commented on hil/hil.py:
```diff
@@ -1,107 +1,99 @@
-from hil_devices.hil_device import HilDevice
-from hil_devices.serial_manager import SerialManager
-from components.component import Component
+# import time
```
delete unused include

AdityaAsGithub commented on hil/hil.py:
```diff
@@ -1,107 +1,99 @@
+PIN_MAP_PATH = "../pin_maps"
 
-PARAMS_PATH = "..\hil_params.json"
+PARAMS_PATH = "../hil_params.json"
```
This isn't in the scope of your pr to fix but we should not be doing this

AdityaAsGithub commented on hil/hil.py:
```diff
@@ -1,107 +1,99 @@
+        # self.global_failed_checks = []
+        # self.global_test_names = []
+        # self.global_check_count = 0 # multiple checks within a test
+        # self.global_test_count = 0
```
put an old copy of this file in the old folder, and delete all this commentedd code.

AdityaAsGithub commented on hil/hil.py:
```diff
@@ -1,107 +1,99 @@
         self.daq_config = utils.load_json_config(os.path.join(config['firmware_path'], config['daq_config_path']), os.path.join(config['firmware_path'], config['daq_schema_path']))
         self.can_config = utils.load_json_config(os.path.join(config['firmware_path'], config['can_config_path']), os.path.join(config['firmware_path'], config['can_schema_path']))
 
+        print(f"PATH: {os.path.join(config['firmware_path'], config['dbc_path'])}")
```
delete

AdityaAsGithub commented on hil/hil.py:
```diff
@@ -1,107 +1,99 @@
 
-    def shutdown(self):
+    def shutdown(self) -> None:
+        print(f"{utils.bcolors.OKCYAN}HIL shutdown START{utils.bcolors.ENDC}")
```
delete prints

AdityaAsGithub commented on hil/hil.py:
```diff
@@ -1,107 +1,99 @@
+    def stop_can(self) -> None:
         if not self.can_bus: return
+        print(f"{utils.bcolors.OKCYAN}HIL stop_can START{utils.bcolors.ENDC}")
+        
```
delete

AdityaAsGithub commented on hil/hil.py:
```diff
@@ -1,107 +1,99 @@
             #     # wait for bus receive to finish
             #     pass
         self.can_bus.disconnect_bus()
+        print(f"{utils.bcolors.OKGREEN}HIL stop_can END{utils.bcolors.ENDC}")
```
delete

AdityaAsGithub commented on hil/hil.py:
```diff
@@ -176,75 +168,80 @@ def get_hil_device_connection(self, board, net):
 
-def signal_int_handler(signum, frame):
+
+def signal_int_handler(signum, frame) -> None:
```
Give another shot

AdityaAsGithub commented on hil/hil_devices/hil_device.py:
```diff
@@ -1,28 +1,30 @@
-import utils
+# import time
+# import serial
+# import serial.tools.list_ports
```
delete

AdityaAsGithub commented on hil/hil_devices/hil_device.py:
```diff
@@ -75,27 +77,30 @@ def get_port_number(self, port_name, mode):
-    def write_gpio(self, pin, value): 
+    def write_gpio(self, pin: int, value: int) -> None: 
         data = [(HIL_CMD_WRITE_GPIO & HIL_CMD_MASK), (pin & HIL_ID_MASK), value]
+        # print(f"write pin {pin} to {value}")
```
delete

AdityaAsGithub commented on hil/hil_devices/hil_device.py:
We need to test this

AdityaAsGithub commented on hil/hil_devices/hil_device.py:
```diff
@@ -75,27 +77,30 @@ def get_port_number(self, port_name, mode):
-        # print(f"write pin {pin} to {value}")
+    def write_dac(self, pin: int, voltage: float) -> None:
+        value = int(voltage * self.volts_to_dac)
+        char_1 = (value >> 8) & 0xFF
+        char_2 = value & 0xFF
```
I don't want floating numbers

AdityaAsGithub commented on hil/hil_devices/hil_device.py:
```diff
@@ -75,27 +77,30 @@ def get_port_number(self, port_name, mode):
+        char_1 = (value >> 8) & 0xFF
+        char_2 = value & 0xFF
+        data = [(HIL_CMD_WRITE_DAC & HIL_CMD_MASK), (pin & HIL_ID_MASK), char_1, char_2]
+        print(f"write pin {pin} to {voltage}V = {value} ({char_1}, {char_2})")
```
debug print

AdityaAsGithub commented on hil/hil_devices/hil_device.py:
```diff
@@ -104,8 +109,13 @@ def read_analog(self, pin):
         data = [(HIL_CMD_WRITE_POT & HIL_CMD_MASK), (pin & HIL_ID_MASK), value]
         #print(f"sending {value} to pin {pin}")
         self.sm.send_data(self.id, data)
+
+    def write_pwm(self, pin: int, value: int) -> None:
+        data = [(HIL_CMD_WRITE_PWM & HIL_CMD_MASK), (pin & HIL_ID_MASK), value]
+        print(f"writing {value} to pin {pin}")
+        self.sm.send_data(self.id, data)
```
It doesn't work we don't need this for now.

AdityaAsGithub commented on hil/hil_devices/hil_device_arduino_micro.json:
I think we should revert for now

AdityaAsGithub commented on hil/hil_devices/hil_device_arduino_mil_pcb.json:
Branch off this branch to save your work and delete these files from this PR

AdityaAsGithub commented on hil/utils.py:
```diff
@@ -1,42 +1,99 @@
+    # }
+
+    
+    # global plot_x_range_sec
+    # plot_x_range_sec = 10
+    
+    # global events
+    # events = []
```
delete

AdityaAsGithub commented on net_maps/mil_col_net_map.csv:
Delete after moving to new branch

AdityaAsGithub commented on net_maps/mil_pcb_net_map.csv:
Delete after moving to new branch

AdityaAsGithub commented on scripts/test.sh:
I like the idea of a script but I think we should do it better - for now just delete i guess

AdityaAsGithub commented on scripts/test_mil_col.py:
delete after moving to new branch

AdityaAsGithub reviewed (commented):
I'll look at the pytest stuff later

LelsersLasers committed:
Deleting old test files

LelsersLasers committed:
Removing old hil checks/testing suite + copying old file

AdityaAsGithub commented on configurations/config_mil_col.json:
move to new branch

LelsersLasers committed:
Type hinting signal_int_handler

AdityaAsGithub commented on configurations/config_mil_pcb.json:
move to new branch

LelsersLasers committed:
Undoing CollTester

LelsersLasers committed:
Removing TODO: type hint comments

LelsersLasers committed:
Cleaning comment

LelsersLasers committed:
Apparently can delete

LelsersLasers committed:
Removing debug prints

LelsersLasers committed:
Undoing windows -> linux file conversions

Should make a new branch/PR with this change (or just use python pages instead...)

LelsersLasers committed:
Delete debug print

LelsersLasers committed:
Removing unused imports

LelsersLasers committed:
Removing debug print

LelsersLasers committed:
Delete debug print

LelsersLasers committed:
Renaming and making better use of MASK constants

LelsersLasers committed:
Remove debug print

LelsersLasers committed:
Delete test.sh

LelsersLasers committed:
Remove unused import

LelsersLasers committed:
Revert

LelsersLasers committed:
Delete unused

LelsersLasers committed:
Removing PWM code

LelsersLasers committed:
Deleting my personal test files from the PR

LelsersLasers committed:
Removing collector plate tests from the PR

LelsersLasers committed:
char != uint8_t

AdityaAsGithub commented on TestBench/TestBench.ino:
```diff
@@ -0,0 +1,194 @@
+			int pin = data[1];
+			int value = data[2];
+			// if (pin < DIGITAL_PIN_COUNT)
+			if (1) {
```
I suggest we fix this because an if(1) is kinda useless.

AdityaAsGithub commented on hil/hil.py:
```diff
@@ -176,75 +155,52 @@ def get_hil_device_connection(self, board, net):
-if __name__ == "__main__":
-    hil = HIL()
-    hil.load_config("config_test.json")
+# if __name__ == "__main__":
```
Add a clarifying comment here to make sure ppl know why this exists. We can remove this when the old folder gets canned.

AdityaAsGithub commented on hil/hil.py:
```diff
@@ -176,75 +155,52 @@ def get_hil_device_connection(self, board, net):
 
-    def can_var(self, board, message_name, signal_name):
+    def can_var(self, board: str, message_name: str, signal_name: str) -> BusSignal:
+        # TODO: not sure if any of the type hints are correct
```
still have this here

AdityaAsGithub commented on hil/hil_devices/hil_device_arduino_micro.json:
```diff
@@ -26,6 +26,6 @@
-   ],
-   "adc_config":{"bit_resolution":10, "reference_v":5.0}
+    ],
+    "adc_config":{"bit_resolution":10, "reference_v":5.0}
```
If you need to repush delete this whitespace

LelsersLasers replied:
Actually, this change I think makes sense. These two lines were tabbed using 3 spaces when every other line uses 4 space tabs (and this makes them 4 spaces)

AdityaAsGithub reviewed (commented):
After these changes I'll tentatively say I approve since we are gonna rewrite the test scripts anyway.

LelsersLasers committed:
Deleting todo

LelsersLasers committed:
Removing unused variable

LelsersLasers committed:
Adding comment

LelsersLasers committed:
Removing if (1)

AdityaAsGithub reviewed (approved):
Approved with the stipulation that the test scripts themselves are getting rewritten soon anyway, so they don't matter for the time being.
